### ISSUE TITLE
Plotting prior and posterior for plot.variance.decomposition

### ISSUE THREAD

**Issue by braczka** [PecanProject/pecan#673]
@dlebauer @mdietze An old issue that I never fully resolved, sorry.  I want to plot both prior and posterior model results onto a single variance decomposition plot as in LeBauer at al. 2013.   I am using the latest VM release 1.4.3.ova.  I updated the source code based upon issue #93 submitted a while back by following commit  `<b685f92>`.    and made appropriate theme system updates.

I can sucessfully run the function:
`<vd.plots <- plot.variance.decomposition(postre.vd$variance.decomposition.output,post.vd$variance.decompoosition.output)>`

but when plotting the results <do.call(grid.arrange, c(vd.plots, ncol = 4))> receive the following error:
`<Error in data.frame(x = c(0, 0), y = c(0, 0), yend = c(0, 120), xend = c(15,  : 
  arguments imply differing number of rows: 2, 15>`

This error is coming from the mapping variable within `<vd.plots$trait.plot$mapping>`.  I do not have enough familiarity with R to troubleshoot but have attached data files.  Not sure its properly merging both sets of posterior data within `<vd.plots>`.   Can send Rdata files if that might help.  Thanks.

**Comment by dlebauer** [comment]
Yes, I've seen this before but forget where. Please post Rdata files and
any other information required for a reproducible example

http://adv-r.had.co.nz/Reproducibility.html

**Comment by braczka** [comment]
@dlebauer Github is telling me it does not support .Rdata type formatted files.  Can I send to you directly or are there ways around this?

**Comment by dlebauer** [comment]
There is dput as illustrated here:
http://adv-r.had.co.nz/Reproducibility.html if it's too big copy paste to
gist.github.com

**Comment by braczka** [comment]
@dlebauer Github is giving me an error when attaching documents:  'Attaching documents requires write permission to this repository'.   Do my user settings need to be changed?

**Comment by dlebauer** [comment]
You can create a gist and post a link

**Comment by mdietze** [comment]
@braczka you could always just email David the files if you can't post them publically

**Comment by braczka** [comment]
files sent to @dlebauer gmail

**Comment by dlebauer** [comment]
@braczka I would really be happy to help you to debug this, but I still can't reproduce your error with the files that you sent.

If I do 

```r
source("~/Downloads/plots.R")
source("~/Downloads/plot_multiple_github.R")
```

I get the error: 

> Error: Incompatible lengths for set aesthetics: size

Thus I am unable to reproduce the bug that you have reported. And at 2500 lines, the example is way to big for me to try to debug.

If you can narrow down the error and provide a [reproducible example](http://adv-r.had.co.nz/Reproducibility.html) of the reported bug then I can take another look.

**Comment by dlebauer** [comment]
One additional bit of information to provide is the output from `traceback()` after the Error occurs. If you post that, it may ring a bell.

**Comment by braczka** [comment]
Github allowed me attach code and `traceback()` this time around.   To recreate bug I did the following:
1) Downloaded PECAN package associated with VM release 1.4.3. Environment included at top of 'plot_multiple_github.R' 
2) Made edits to the PECAN uncertainty file `<plots.R>` according to LeBauer, Github issue #93.   Edited `plot.R` is attached here. 
3) `>./pecan/scripts/build.sh`
3) `>source('~/plot_multiple_github.R')`

`plot_multiple_gitbhub.R.` calls upon the function `plot.variance.decomposition` function which is included within `plots.R`.   The only active lines of code (besides the large data files created by `dput` is Lines 2412-2431 within `plot_multiple_github.R`, and Lines 71-216 within `plots.R` which 
is the edited `plot.variance.decomposition` function


[traceback.txt](https://github.com/PecanProject/pecan/files/31328/traceback.txt)
[plot_multiple_github.txt](https://github.com/PecanProject/pecan/files/31329/plot_multiple_github.txt)
[plots.txt](https://github.com/PecanProject/pecan/files/31330/plots.txt)
